src/Completion.hx

```diff
@@ -2,46 +2,58 @@ package;
 import haxe.xml.Fast;
 import js.Browser;
 import js.html.DivElement;
+import js.html.SpanElement;
 import js.codemirror.*;
+import api.Completion.CompletionItem;
 /**
  * ...
  * @author AS3Boyan
  */
 
 //Ported from HIDE, adjusted for try-haxe
- 
-typedef CompletionItem = 
-{
-	@:optional
-	var d:String;
-	@:optional
-	var t:String;
-	var n:String;
-}
     
 typedef CompletionData =
 {
-	var text:String;
+	@:optional var text:String;
 	@:optional var displayText:String;
 	@:optional var hint:CodeMirror->Dynamic->CompletionData->Void;
+	@:optional var info:CompletionData->Dynamic;
+	@:optional var className:String;
 }
  
 class Completion
 {
-	static var list:Array<CompletionData>;
-	static var editor:CodeMirror;
-	static var word:EReg;
-	static var range:Int;
-	static var cur:CodeMirror.Pos;
-	static private var end:Int;
-	static private var start:Int;
-	static var WORD:EReg = ~/[A-Z]+$/i;
-	static var RANGE = 500;
-	public static var curWord:String;
-	public static var completions:Array<CompletionItem> = [];
-	static var backupDocValue:String;
+	var list:Array<CompletionData>;
+	var editor:CodeMirror;
+	var word:EReg;
+	var range:Int;
+	var cur:CodeMirror.Pos;
+	private var end:Int;
+	private var start:Int;
+	var WORD:EReg = ~/[A-Z]+$/i;
+	var RANGE = 500;
+	public var curWord:String;
+	public var completions:Array<CompletionItem> = [];
+	var backupDocValue:String;
 	
-	public static function registerHelper() 
+	static var instance:Completion;
+	
+	public static function get()
```

Comment:
Not a big fan of singletons :|
How about making just using `new`, retaining the `Completion` instance in `Editor`, and pass it to `FunctionParameterHelper`, `ColorPreview`, etc  ?